libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1472,13 +1484,61 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		m_context.appendConditionalInvalid();
 
 		if (_operator == Token::Div)
-			m_context << (c_isSigned ? Instruction::SDIV : Instruction::DIV);
+		{
+			if (fixedType)
+			{
+				// Division would cancel out factor, this gives the numerator an extra factor so it remains in result
+				m_context << u256(pow(bigint(10), fixedType->fractionalDigits()))
+						  << Instruction::MUL;
+			}
+			m_context << (isSigned ? Instruction::SDIV : Instruction::DIV);
+		}
 		else
-			m_context << (c_isSigned ? Instruction::SMOD : Instruction::MOD);
+			m_context << (isSigned ? Instruction::SMOD : Instruction::MOD);
 		break;
 	}
 	case Token::Exp:
-		m_context << Instruction::EXP;
+		if (fixedType)
+		{
+			u256 factor = u256(pow(bigint(10), fixedType->fractionalDigits()));
+
+			// stack: exp base
+
+			// Test for negative exponent
+			m_context << Instruction::DUP2
+					  << u256(0)
+					  << Instruction::GT;
+			m_context.appendConditionalInvalid();
+
+			// stack: exp base
+
+			// Test for non-integer exponent
+			m_context << Instruction::DUP2 		// exp base exp
```

Comment:
You can (and actually should) use `appendInlineAssembly()` here to increase readability by some orders of magnitude :)

Reply:
I'll do that and convert it to use a loop.

Reply:
Why would you use a loop to decrease overflows? (I hope you didn't mean assembly loop and meant something else).